### DISCUSSION TITLE
Home assistant Cover Sensor

### DISCUSSION THREAD

**Discussion by timknowlden** [omersiar/esp-rfid#576] [Q&A] [answered]
Has anyone integrated into HA as an MQTT Cover?

I am using this for my garage door and have a door sensor so I would like to be able to have this as a cover entity. Autodiscover isn't working as far as I can tell.

**Comment by matjack1** [comment]
hey @timknowlden, can you provide a copy of your settings? You can export them from the web UI. So I can have a look and better understand what's going on. Autodiscovery should work :)

**Comment by timknowlden** [comment]
[esp-rfid-settings-2.json.zip](https://github.com/esprfid/esp-rfid/files/10475383/esp-rfid-settings-2.json.zip)
@matjack1 These are my settings, I have tried to change back the rtime to 300ms, but I have saved and rebooted 5 times now without affect.

**Comment by matjack1** [comment]
@timknowlden you should enable this setting in the MQTT section:

![Schermata 2023-01-22 alle 21 40 25](https://user-images.githubusercontent.com/65402/213939307-20d69248-7260-4417-b714-29aa0fedee76.jpg)

from the settings that you have shared this option is not set, can you please try again with that?

**Comment by timknowlden** [comment, marked as answer]
Hi @matjack1 it seems now I have flashed using the UART pins with the latest dev bin instead of trying OTA from existing version that all of my issues have been resolved including the relay trigger on start. 

I will continue to test and now that the home assistant auto discovery is working I can adjust my node red flows to suit the new sensors and see if I can build a cover template with these to suit a garage door.

I will also add my doorbell button and all being well should be able to install this soon.

**Reply by matjack1** [reply]
Nice!
